runtime/compiler/control/CompilationRuntime.hpp

```diff
@@ -674,12 +674,46 @@ class CompilationInfo
 #if defined(J9VM_OPT_JITSERVER)
       TR_ASSERT_FATAL(!TR::CompilationInfo::getStream(), "not yet implemented for JITServer");
 #endif /* defined(J9VM_OPT_JITSERVER) */
-      value = (value << 1) | J9_STARTPC_NOT_TRANSLATED;
-      if (value < 0)
-          value = INT_MAX;
-      method->extra = reinterpret_cast<void *>(static_cast<intptr_t>(value));
+#if defined(J9VM_OPT_MICROJIT)
+      TR::Options *options = TR::Options::getJITCmdLineOptions();
+      /* Setting MicroJIT invocation count to 0
+         causes some methods to not get compiled.
+         Hence we generally set it to a non-zero value.
+         20 has been proven to be a good estimate for count value
+         for similar template-based JITs.
+       */
+      if (options->_mjitEnabled && value)
+         {
+         return;
+         }
+      else
+#endif
+         {
+         value = (value << 1) | J9_STARTPC_NOT_TRANSLATED;
+         if (value < 0)
+            value = INT_MAX;
+         method->extra = reinterpret_cast<void *>(static_cast<intptr_t>(value));
+         }
       }
 
+#if defined(J9VM_OPT_MICROJIT)
+   static void setInitialMJITCountUnsynchronized(J9Method *method, int32_t mjitThreshold, int32_t trCount, J9JITConfig *jitConfig, J9VMThread *vmThread)
+      {
+      if (TR::Options::getJITCmdLineOptions()->_mjitEnabled)
+         {
+         intptr_t value = (intptr_t)((mjitThreshold << 1) | J9_STARTPC_NOT_TRANSLATED);
+         if (trCount < value)
+            {
+            value = trCount;
+            TR_J9VMBase *fe = TR_J9VMBase::get(jitConfig, vmThread);
+            U_8 *extendedFlags = fe->fetchMethodExtendedFlagsPointer(method);
+            *extendedFlags = (*extendedFlags) | J9_MJIT_FAILED_COMPILE;
```

Comment:
Here we are setting the count to compile with TR, and hence signalling failure with MJIT.
We want to ensure that if MJIT count is higher than the AOT or TR count then we don't override it. So in that case, it would compile that method with TR or AOT and not MJIT, and that flag is how we know whether to compile with MJIT or TR at the decision point in the compile path.